### PR TITLE
KTOR-4141 KTOR-2143 Fixed inconsistency between server engines when determining port/host of an incoming request

### PR DESCRIPTION
**Pull request by rsinukov**
https://youtrack.jetbrains.com/issue/KTOR-430
https://youtrack.jetbrains.com/issue/KTOR-2501
https://youtrack.jetbrains.com/issue/KTOR-4141
https://youtrack.jetbrains.com/issue/KTOR-2143

### CODE REVIEW

File: ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt [hunk 1]
```diff
@@ -23,13 +23,40 @@ public interface RequestConnectionPoint {
     /**
      * Request port, for example 80 or 443
      */
+    @Deprecated("Use localPort or serverPort instead")
     public val port: Int
 
+    /**
+     * Request port on which the request was received, for example 80 or 443
```
Comment (e5l) [inline ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt:30]: I'm not sure that I understand the doc. I would say it's `The port used to accept the client connection. It's usually specified in the client request`

File: ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt [hunk 2]
```diff
@@ -23,13 +23,40 @@ public interface RequestConnectionPoint {
...
     /**
      * Request host, useful for virtual hosts routing
      */
+    @Deprecated("Use localHost or serverHost instead")
     public val host: String
 
+    /**
+     * Request host on which the request was received, is useful for virtual hosts routing
+     */
+    public val localHost: String
```
Comment (e5l) [inline ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt:48]: is this hostname only?

**Review by e5l — APPROVED** [review]
LGTM 

@andreyaksenov, could you check docs?
@rsinukov, please run the space build after merging

File: ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt:30]: I took the same terminology and naming as in Jetty

File: ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt (hunk 2 above)
Comment (rsinukov) [reply ktor-http/common/src/io/ktor/http/RequestConnectionPoint.kt:48]: yes, port is separate property

**Pull request merged**